app/src/main/java/org/simple/clinic/drugs/EditMedicinesUiRenderer.kt

```diff
@@ -10,14 +10,43 @@ class EditMedicinesUiRenderer(private val ui: EditMedicinesUi) : ViewRenderer<Ed
 
   override fun render(model: EditMedicinesModel) {
     if (model.prescribedDrugs != null && model.protocolDrugs != null)
-      renderPrescribedProtocolDrugs(model.prescribedDrugs, model.protocolDrugs)
+      renderPrescribedProtocolDrugs(model, model.prescribedDrugs, model.protocolDrugs)
   }
 
-  private fun renderPrescribedProtocolDrugs(prescribedDrugs: List<PrescribedDrug>, protocolDrugs: List<ProtocolDrugAndDosages>) {
+  private fun renderPrescribedProtocolDrugs(
+      model: EditMedicinesModel,
+      prescribedDrugs: List<PrescribedDrug>,
+      protocolDrugs: List<ProtocolDrugAndDosages>
+  ) {
     val prescribedProtocolDrugs = prescribedDrugs.filter { it.isProtocolDrug }
-    val isAtLeastOneCustomDrugPrescribed = prescribedDrugs.any { it.isProtocolDrug.not() }
+    val isAtLeastOneCustomDrugPrescribed = prescribedDrugs.any { prescribedDrug -> model.isProtocolDrug(prescribedDrug).not() }
+
+    val protocolDrugSelectionItems = protocolDrugSelectionItems(protocolDrugs, prescribedProtocolDrugs, isAtLeastOneCustomDrugPrescribed)
+    val customPrescribedDrugItems = customPrescribedDrugItems(model, prescribedDrugs)
+    val drugsList = protocolDrugSelectionItems + customPrescribedDrugItems
+
+    ui.populateDrugsList(drugsList)
```

Comment:
Can we add a UI renderer test to verify that a `PrescribedDrug` which was added as a `ProtocolDrug` initially shows up as a non-protocol drug if the protocol drug was removed?

Reply:
Done